### DISCUSSION TITLE
PyQtGraph equivalent to matplotlib.patches.ConnectionPatch

### DISCUSSION THREAD

**Discussion by kylefhartzenberg** [pyqtgraph/pyqtgraph#3091] [Q&A]
Is there an equivalent to matplotlib's [patches.ConnectionPatch](https://matplotlib.org/stable/api/_as_gen/matplotlib.patches.ConnectionPatch.html) in PyQtGraph to produce something similar to the following?

![md-example](https://github.com/pyqtgraph/pyqtgraph/assets/113495179/e09b10a9-d07c-4b35-ac98-945d5c23f049)

It is essentially a visualisation of the mapping between points on the horizontal axes of both images. 
If something is not readily available in PyQtGraph, is there a recommend way to go about producing such a visualisation?

Image source: Herman Kamper, Dynamic Time Warping lecture, published Jul 11, 2021.
Source code to produce the above image can be found here:
https://github.com/kamperh/lecture_dtw_notebook/blob/main/dtw.ipynb

**Comment by pijyoi** [comment]
https://pyqtgraph.readthedocs.io/en/latest/api_reference/widgets/matplotlibwidget.html

**Reply by kylefhartzenberg** [reply]
Thank you. This may help those that don't mind adding `matplotlib` as a dependency, however, solutions or ideas without such a dependency are preferable.

**Comment by j9ac9k** [comment]
That would be a new plot type to add, and absent a PR that is robust enough
I don’t see that happening.

Based on the screenshot you sent this can likely be done with two image
items added to a ViewBox and a PlotCurveItem with connect=‘pairs’, but to
figure out the positioning of the image items and the lines, you’ll be on
your own.
